bellman/src/main.rs

```diff
@@ -0,0 +1,315 @@
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+extern crate bellman;
+extern crate pairing;
+extern crate rand;
+extern crate ff;
+use bellman::{Circuit, ConstraintSystem, SynthesisError, Variable, LinearCombination};
+use pairing::{Engine};
+use std::fs::File;
+use std::io::{BufReader, BufRead};
+use std::time::{Instant};
+
+use ff::{
+    Field,
+    PrimeField,
+};
+
+struct ParsedLine<E: Engine> {
+    index: usize,
+    lc: LinearCombination<E>,
+    debug: String,
+    result: E::Fr,
+}
+
+struct Assignment<E: Engine> {
+    variable: Variable,
+    constrained: bool,
+    value: E::Fr,
+}
+
+fn empty_line<E:Engine>(index: usize) -> ParsedLine<E> {
+    ParsedLine::<E>{
+        index:index, 
+        lc: LinearCombination::zero(), 
+        debug: "".to_string(), 
+        result: E::Fr::zero(),
+    }
+}
+
+fn parse_line<E: Engine>(line: &String, variables: &mut Vec<Assignment<E>>, negate: bool) -> ParsedLine<E> {
+    let parts: Vec<&str> = line.split_whitespace().collect();
+    let row: usize = parts[0].parse().unwrap();
+    let column: usize = parts[1].parse().unwrap();
```

Comment:
I am concerned that rows and columns here are reversed. I have been interpreting this as `column row value`

Reply:
I think it depends what you think of as row/column. From: https://github.com/pepper-project/pequin/blob/master/fileformats.txt

```
E.g. if there is a line in a *.matrix_a file:

96 5 2

That means the entry at 96th row, 5th column is 2.
```

So they define it as `row column coeff`